Util/Xamarin.Build.Download/source/Xamarin.Build.Download/XamarinBuildAndroidAarRestore.cs

```diff
@@ -4,19 +4,53 @@
 using System.Linq;
 using Microsoft.Build.Framework;
 using System.Xml.Linq;
+using Microsoft.Build.Utilities;
+using System.Collections.Generic;
 
 namespace Xamarin.Build.Download
 {
 	public class XamarinBuildAndroidAarRestore : XamarinBuildAndroidResourceRestore
 	{
 		public bool FixAndroidManifests { get; set; } = true;
 
-		protected override Stream LoadResource (string resourceFullPath, string assemblyName)
+		[Output]
+		public ITaskItem [] AarProguardConfiguration { get; set; }
+
+		string proguardIntermediateOutputPath = null;
+
+		public override bool Execute ()
```

Comment:
So I think this is actually the case, the base will skip this. I think I'm going to need to move this bit into its own task that is always run.